packages/react-vapor/src/components/sideNavigation/SideNavigationMenuSection.tsx

```diff
@@ -1,63 +1,86 @@
+import * as classNames from 'classnames';
 import * as React from 'react';
-import {SlideY} from '../../animations/SlideY';
+import * as _ from 'underscore';
+
+import {Collapsible} from '../collapsible/Collapsible';
 import {Svg} from '../svg/Svg';
-import {ISideNavigationHeaderProps, SideNavigationHeader} from './SideNavigationHeader';
-import {SideNavigationLoadingHeader} from './SideNavigationLoadingHeader';
+import {ISideNavigationHeaderProps} from './SideNavigationHeader';
 
-export interface ISideNavigationSectionOwnProps {
+export interface ISideNavigationSectionProps extends SideNavigationHeaderProps {
+    /**
+     * @deprecated pass those in props directly
+     */
     header?: ISideNavigationHeaderProps;
     expandable?: boolean;
-}
-
-export interface ISideNavigationSectionStateProps {
     expanded?: boolean;
-}
-
-export interface ISideNavigationSectionDispatchProps {
     onClick?: () => void;
 }
 
-export interface ISideNavigationSectionProps
-    extends ISideNavigationSectionOwnProps,
-        ISideNavigationSectionStateProps,
-        ISideNavigationSectionDispatchProps {}
-
-const arrowClassName = 'collapsible-arrow icon fill-white transparency-4';
-
-export class SideNavigationMenuSection extends React.Component<ISideNavigationSectionProps> {
-    private handleClick() {
-        if (this.props.onClick) {
-            this.props.onClick();
-        }
-    }
+export interface SideNavigationHeaderProps {
+    title?: React.ReactNode;
+    svgName?: string;
+    svgClass?: string;
+    customIcon?: React.ReactNode;
+    onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => void;
+}
 
-    render() {
-        return (
-            <div className="block navigation-menu-section">
-                {this.buildHeader()}
-                <SlideY in={!(this.props.expandable && !this.props.expanded)} timeout={350}>
-                    <div className="navigation-menu-section-items">{this.props.children}</div>
-                </SlideY>
-            </div>
+const SideNavigationHeader: React.FunctionComponent<SideNavigationHeaderProps> = (props) => {
+    let icon: React.ReactNode = null;
+    if (props.customIcon) {

```

Comment:
🤔 what do you think about changing it a little to have a `const icon` instead? 

```
const icon = props.customIcon || (props.svgName ? <this.SvgIcon /> : <span className="navigation-menu-section-header-icon" />
```

Reply:
So you would convert `SideNavigationHeader` in a class component to define `<this.SvgIcon />` ?

Reply:
Oh hum, definitely don't want to convert :)

What about a local functional component? 🤔

```
const SvgIcon = () => <Svg ... />;
const icon = props.customIcon || (props.svgName ? <SvgIcon /> : <span className="navigation-menu-section-header-icon" />
```

I mean, I find it slightly easier to follow, but your original code is not that bad either ;)

Reply:
Aight good idea I'll convert it to something like this

```tsx
const HeaderIcon: React.FunctionComponent<SideNavigationHeaderProps> = ({svgName, svgClass}) =>
    svgName ? (
        <Svg
            svgName={svgName}
            svgClass={classNames('navigation-menu-section-header-icon icon mod-lg transparency-3 fill-white', svgClass)}
        />
    ) : (
        <span className="navigation-menu-section-header-icon" />
    );
```